src/java.base/share/classes/sun/security/ssl/JsseJce.java

```diff
@@ -155,6 +181,22 @@ static RSAPublicKeySpec getRSAPublicKeySpec(PublicKey key) {
         }
     }
 
+    /**
+     * Checks if a particular signature algorithm exists.
+     *
+     * @param algorithm The algorithm which we will attempt to instantiate to check for presence.
+     * @return true if the signature algorithm is found, false otherwise.
+     */
+    private static boolean isSignatureAlgorithmAvailable(String algorithm) {
+        try {
+            // Attempt to create a Cipher instance with the specified algorithm
+            Signature.getInstance(algorithm);
+            return true;
+        } catch (NoSuchAlgorithmException | javax.crypto.NoSuchPaddingException e) {
+            return false;
+        }
+    }
```

Comment:
This method can be within `EcAvailability` (it's only used there).

I think the description should say "available" rather than "exists" to match the name.

By convention `@` taglets do not end with a period, but the comment in the method should.

`NoSuchPaddingException` is imported, so need not be fully-qualified here.

Reply:
Moved the method to the `EcAvailability`

Changed the description to be `available` instead of exists.

Updated the taglets to no longer have a period at the end. Updated the comment in the method to have a period.

I found out during compilation i dont need `NoSuchPaddingException`. This was removed.